### ISSUE TITLE
Features/Ideas/Improvements

### ISSUE THREAD

**Issue by 777arc** [IQEngine/IQEngine#1]
### S-Tier Features
- After setting selection cursors, right click (or some other way) to save the samples to a new file you can download
- Wavelet transforms!  Including Readme/tutorial about why wavelets are useful, inspired by  https://www.mathworks.com/help/wavelet/ug/time-frequency-analysis-and-continuous-wavelet-transform.html
- Zooming in in time by decimating (instead of just changing fft size), less data gets fetches but its not continuous anymore
- Selection cursors should show up in minimap, and they shouldn't reset every time you show/hide them
- Ability to tune-filter-decimate a section after choosing the region of freq of interest (and time-domain selection), and then you can download the resulting samples

### B-Tier Features
- Detect if the annotation rectangle text/label intersects with another one, and if so, put one at the bottom of the rectangle instead of top
- Ability to use a polyphase filterbank channelizer in place of FFT for better time resolution anaylsis
- Ability to convert non-sigmf files (e.g. pcap of difi) to SigMF, both locally and as an Azure Function
- Basic cyclostationary processing, e.g. brute force SCF
- Ability to link someone to a specific repo + recording, and even a specific sample_start would be cool
- Ability to download spectrogram images from webapp, like a one-button screenshot that gives you a png of whatever is on your spectrogram viewer at the time

### Misc Cleanups or Bugs
- On spectrogram page we are missing units on the scales
- So many spots where we have to divide by bytes per sample, would be nice to normalize to int16s or float32s and never have to deal with it in the code other than when the bytes are first downloaded

### Azure Oriented
- Create an ARM template that spins up the storage account, python azure functions, cosmosdb, an app service plan
-Autodeploy for our azure functions

**Comment by ianfinley89** [comment]
- It would be great if annotations could propagate through the frequency plot.

- Could annotations also be added to the IQ data in the form of multi colors with a legend?

- We can add cursors for a span of time, could we add cursors for a span of frequency as well?

**Comment by 777arc** [comment]
Yeah we definitely need freq domain cursors, as well as the calc of delta-t and delta-f (plus inverse) for time and freq

**Comment by 777arc** [comment]
updated the list 6-30-23

**Comment by 777arc** [comment]
updated

**Comment by 777arc** [comment]
updated

**Comment by johnmarcovici** [comment]
Option to transpose the spectro? I love seeing time on x axis, freq on y.

**Comment by johnmarcovici** [comment]
Units and sigfigs on spectro. Freq in MHz --> 2 sigfigs is typical
What units on time? Prefer usec typically but label it whatever it may be

<img width="1718" height="603" alt="Image" src="https://github.com/user-attachments/assets/01847690-e6b1-4768-a0c1-2fa54a018d78" />

**Comment by johnmarcovici** [comment]
What are the delta's supposed to mean here?
For legibility / intelligibility maybe this can read
Num samples = blah, time span = blah
or something similarly unambiguous
Also, control the sigfigs on time.

<img width="1073" height="479" alt="Image" src="https://github.com/user-attachments/assets/e8a17328-364a-4d9d-838e-f78106200c4d" />

**Comment by 777arc** [comment]
Great suggestions!  Any chance you have javascript xp? A lot of the suggestions are relatively simple tweaks, eg here's the logic that produces the time domain selector text https://github.com/IQEngine/IQEngine/blob/main/client/src/pages/recording-view/components/time-selector.tsx

**Comment by johnmarcovici** [comment]
I don't have a Javascript xp... Let me look into that. Presuming that would
enable me to toss up some pr's for small stuff like this?

**Comment by 777arc** [comment]
Yeah if you wanted to play with tweaking things, it's actually pretty easy once you have a local dev environment set up, especially with AI, because you run a local instance of IQEngine and any javascript changes are immediately updated.  It's best on ubuntu so I use WSL with Ubuntu 24, and here are the steps https://iqengine.org/docs/new-developer-guide happy to help you set it up locally.

**Comment by johnmarcovici** [comment]
OK let me look into.
